plugins/package-managers/node/src/main/kotlin/Npm.kt

```diff
@@ -246,8 +240,13 @@ open class Npm(
 
         logger.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        val nodeModulesFiles = nodeModulesDir.walk().maxDepth(modulesSearchDepth).filter {
-            it.name == "package.json" && isValidNodeModulesDirectory(nodeModulesDir, nodeModulesDirForPackageJson(it))
+        val visitedDirs = mutableSetOf<File>()
```

Comment:
commit-message: Is this really a generalization for the mentioned max depth work around, or is it a different work around (or maybe even a proper solution) for the same underlying problem.

Also, would it make sense to link the commit for that Pnpm work around. I wasn't unable to find this quickly.

Reply:
> or maybe even a proper solution

I believe it's indeed a proper solution. The `maxDepth`-based approach did not really break the cycles, but just followed them until a maximum depth (that was set just so that all packages are found in the created directory hierachry). The solution in this PR really breaks the cycle.

> Also, would it make sense to link the commit for that Pnpm work around. I wasn't unable to find this quickly.

The work-around was there from the start, when PNPM support was introduced in 18d8903f43f4a8a7b77e87f4042fd38514a4436e. I've mentioned it in the commit message.